R/fit_drc_4p.R

```diff
@@ -206,17 +247,16 @@ fit_drc_4p <- function(data,
     vector_add <- c(vector[1] + 1, vector[2] - 1)
     vector_add_rev <- rev(vector_add)
 
-    concentrations <- sort(unique(dplyr::pull(data_prep, {{ dose }})))
-
     filter_completeness <- data_prep %>%
       dplyr::group_by({{ grouping }}, {{ dose }}) %>%
       dplyr::mutate(enough_replicates = sum(!is.na({{ response }}))
-      >= n_replicates_completeness) %>%
+      >= n_replicate_completeness) %>%
```

Comment:
I think the data manipulation from line 250-390 is quite difficult to read:
- i would try to reduce the number of mutate()-functions and combine these manipulations into one function (it will also enhance the performance as the pipe %>% will copy the data each time)
as for instance here: data <- data %>%
  dplyr::mutate(
    # Use coalesce to replace NA values in pval_vector columns with TRUE
    pval_vector = dplyr::coalesce(pval_vector, TRUE),
    pval_vector_rev = dplyr::coalesce(pval_vector_rev, TRUE),
    pval_vector_add = dplyr::coalesce(pval_vector_add, TRUE),
    pval_vector_add_rev = dplyr::coalesce(pval_vector_add_rev, TRUE),
    # Use coalesce to replace NA values in mean_vector columns with 0
    mean_vector = dplyr::coalesce(mean_vector, 0),
    mean_vector_rev = dplyr::coalesce(mean_vector_rev, 0),
    mean_vector_add = dplyr::coalesce(mean_vector_add, 0),
    mean_vector_add_rev = dplyr::coalesce(mean_vector_add_rev, 0)
  )

Reply:
I agree, it is a bit difficult to read. I reduced the number of `mutate()`. And replaced the mentioned code using `replace_na()`:

```
data %>%
      tidyr::replace_na(list("pval_vector" = TRUE,
                             "pval_vector_rev" = TRUE,
                             "pval_vector_add" = TRUE,
                             "pval_vector_add_rev" = TRUE,
                             "mean_vector" = 0,
                             "mean_vector_rev" = 0,
                             "mean_vector_add" = 0,
                             "mean_vector_add_rev" = 0)) 
```